src/components/ebay-pagination/test/test.browser.js

```diff
@@ -0,0 +1,108 @@
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-previous', spy);
+            testUtils.triggerEvent(previousButton, 'click');
+        });
+
+        test('then it emits the marko event called pagination-previous', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+    describe('when the next button is clicked', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-next', spy);
+            testUtils.triggerEvent(nextButton, 'click');
+        });
+
+        test('then it emits the marko event called pagination-next', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+
+    describe('when the page number is clicked', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-select', spy);
+            testUtils.triggerEvent(pageItem, 'click');
+        });
+
+        test('then it emits the marko event called pagination-select', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el', 'value']);
+        });
+    });
+
+    describe('when the previous button is activated through keydown', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-previous', spy);
+            testUtils.triggerEvent(previousButton, 'keydown', 32);
+        });
+
+        test('then it emits the marko event called pagination-previous', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+    describe('when the next button is activated through keydown', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-next', spy);
+            testUtils.triggerEvent(nextButton, 'keydown', 32);
+        });
+
+        test('then it emits the marko event called pagination-next', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+
+    describe('when the page number is activated through keydown', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-select', spy);
+            testUtils.triggerEvent(pageItem, 'keydown', 32);
+        });
+
+        test('then it emits the marko event called pagination-select', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el', 'value']);
+        });
+    });
+});
```

Comment:
Would be good to test the resize behavior too, something like this:
https://github.com/eBay/ebayui-core/pull/81/files#diff-312b1af3550afda88d75e63a6c6bfcc8R31

Reply:
Yes! I will add the test cases.